applications/vessel-history/src/routes/routes.hook.ts

```diff
@@ -1,10 +1,86 @@
 import { useSelector, useDispatch } from 'react-redux'
-import { useCallback } from 'react'
+import { useCallback, useEffect } from 'react'
+import { parse } from 'qs'
+import { useLocalStorage } from '@globalfishingwatch/react-hooks'
+import GFWAPI, { ACCESS_TOKEN_STRING } from '@globalfishingwatch/api-client'
+import { parseWorkspace } from '@globalfishingwatch/dataviews-client'
 import { QueryParams } from 'types'
-import { selectCurrentLocation, selectLocationPayload } from 'routes/routes.selectors'
+import {
+  selectCurrentLocation,
+  selectLocationPayload,
+  selectLocationType,
+} from 'routes/routes.selectors'
 import { ROUTE_TYPES } from './routes'
 import { updateLocation } from './routes.actions'
 
+export const CALLBACK_URL_KEY = 'CallbackUrl'
+export const CALLBACK_URL_PARAM = 'callbackUrlStorage'
+
+export const setRedirectUrl = () => {
+  window.localStorage.setItem(CALLBACK_URL_KEY, window.location.toString())
+}
+
+export const getLoginUrl = () => {
+  const { origin, pathname } = window.location
+  return GFWAPI.getLoginUrl(`${origin}${pathname}?${CALLBACK_URL_PARAM}=true`)
+}
+
+export const redirectToLogin = () => {
+  setRedirectUrl()
+  window.location.href = getLoginUrl()
+}
+
+export const useLoginRedirect = () => {
```

Comment:
why not having this hook into the package too? as it is almost the same than the fishing one and we need to add support configuration for the [storage keys](https://github.com/GlobalFishingWatch/frontend/pull/1099/files#diff-b2b39c93fd82e82cff2c3cdf34bfc02ce52eb519af2b8dbe838e4b30ec5f9947R16) WDYT ?

Reply:
It makes sense so we reduce the duplicated code. I'll move it